### PR TITLE
Make React UI the default, keep old UI under /classic

### PR DESCRIPTION
**Pull request by juliusv**
The React app's assets are now served under /assets, while all old
custom web assets (including the ones for console templates) are now
served from /classic/static.

I tested different combinations of --web.external-url and
--web.route-prefix with proxies in front, and I couldn't find a problem
yet with the routing. Console templates also still work.

While migrating old endpoints to /classic, I noticed that /version was
being treated like a lot of the old UI pages, with readiness check
handler in front of it, etc. I kept it in /version and removed that
readiness wrapper, since it doesn't seem to be needed for that endpoint.

Signed-off-by: Julius Volz <julius.volz@gmail.com>

### CODE REVIEW

**Comment by juliusv** [comment]
@roidelapluie @brian-brazil

**Review by bwplotka — APPROVED** [review]
It looks good to me, but another, frontend eyes would be nice here :+1: 

Thanks!

**Comment by juliusv** [comment]
Pfft, got a weird unrelated test error, retrying:

```
goleak: Errors on successful test run: found unexpected goroutines:
[Goroutine 15531 in state chan send, with github.com/prometheus/prometheus/tsdb.(*SegmentWAL).cut.func2 on top of the stack:
goroutine 15531 [chan send]:
github.com/prometheus/prometheus/tsdb.(*SegmentWAL).cut.func2(0xc0001142c0)
	/home/circleci/project/tsdb/wal.go:600 +0xc5
created by github.com/prometheus/prometheus/tsdb.(*SegmentWAL).cut
	/home/circleci/project/tsdb/wal.go:599 +0x16c
]
FAIL	github.com/prometheus/prometheus/tsdb	185.399s
```

File: Makefile [hunk 1]
```diff
@@ -15,10 +15,11 @@
 DOCKER_ARCHS ?= amd64 armv7 arm64 ppc64le s390x
 
 REACT_APP_PATH = web/ui/react-app
-REACT_APP_SOURCE_FILES = $(wildcard $(REACT_APP_PATH)/public/* $(REACT_APP_PATH)/src/* $(REACT_APP_PATH)/tsconfig.json)
+REACT_APP_SOURCE_FILES = $(shell find $(REACT_APP_PATH)/public/ $(REACT_APP_PATH)/src/ $(REACT_APP_PATH)/tsconfig.json)
```
Comment (roidelapluie) [inline Makefile:18]: Why this change?

Comment (juliusv) [reply Makefile:18]: I got annoyed that changes in the React app frequently didn't get picked up by make because the existing dependency wasn't recursive (we have many subdirs). And `**` is a bit arcane, so I decided to go the `find` way, seems most robust in finding everything.

**Comment by juliusv** [comment]
Since two people have at least had a look and there's one approval, I'll allow myself to merge this :)

**Comment by pathcl** [comment]
is there a way to make classic ui default on > 2.23 ? it's working way slower but it might be me or my browser

**Comment by roidelapluie** [comment]
There is no way , but feel free to open an issue about the exact problem you are experiencing and your scale so we can determine if that's a bug.
